easybuild/easyblocks/m/mesa.py

```diff
@@ -40,42 +42,78 @@ class EB_Mesa(MesonNinja):
     """Custom easyblock for building and installing Mesa."""
 
     def __init__(self, *args, **kwargs):
-        """Constructor for custom Mesa easyblock: figure out which vales to pass to swr-arches configuration option."""
+        """Constructor for custom Mesa easyblock: figure out which values to pass to swr-arches configuration option."""
 
         super(EB_Mesa, self).__init__(*args, **kwargs)
 
+        self.gallium_configopts = []
+
+        # Check user-defined Gallium drivers
+        gallium_drivers = self.get_configopt_value('gallium-drivers')
+
+        if not gallium_drivers:
+            # Add appropriate Gallium drivers for current architecture
+            arch = get_cpu_architecture()
+            arch_gallium_drivers = {
+                'x86_64': ['swrast', 'swr'],
+                'POWER': ['swrast'],
+            }
+            if arch in arch_gallium_drivers:
+                gallium_drivers = arch_gallium_drivers[arch]
+                # Add configopt for additional Gallium drivers
+                self.gallium_configopts.append('-Dgallium-drivers=' + ','.join(gallium_drivers))
+
+        self.log.debug('Gallium driver(s) included in the installation: %s' % ', '.join(gallium_drivers))
+
         self.swr_arches = []
 
-        if 'swr-arches' not in self.cfg['configopts']:
-            # set cpu features of SWR for current architecture (only on x86_64)
-            if get_cpu_architecture() == X86_64:
+        if 'swr' in gallium_drivers:
+            # Check user-defined SWR arches
+            self.swr_arches = self.get_configopt_value('swr-arches')
+
+            if not self.swr_arches:
+                # Set cpu features of SWR for current micro-architecture
                 feat_to_swrarch = {
                     'avx': 'avx',
                     'avx1.0': 'avx',  # on macOS, AVX is indicated with 'avx1.0' rather than 'avx'
                     'avx2': 'avx2',
                     'avx512f': 'skx',  # AVX-512 Foundation - introduced in Skylake
                     'avx512er': 'knl',  # AVX-512 Exponential and Reciprocal Instructions implemented in Knights Landing
                 }
-                # determine list of values to pass to swr-arches configuration option
+                # Determine list of values to pass to swr-arches configuration option
                 cpu_features = get_cpu_features()
                 self.swr_arches = sorted([swrarch for feat, swrarch in feat_to_swrarch.items() if feat in cpu_features])
+                # Add configopt for additional SWR arches
+                self.gallium_configopts.append('-Dswr-arches=' + ','.join(self.swr_arches))
+
+            self.log.debug('SWR Gallium driver will support: %s' % ', '.join(self.swr_arches))
+
+    def get_configopt_value(self, configopt_name):
```

Comment:
Can you rename to `get_configopt_values`?  Or better: Just copy my function, there are more issues below.

Reply:
I took most of your function

Reply:
Looks good, best from both :+1:

The name is still remaining. If it returns values (list) it should be named so, shouldn't it? Just in case we need/have something that doesn't split the value into a list

Reply:
I would expect a function called `get_configopt_values` to return all values from `configopts`, not just the last one assigned.